cluster/examples/kubernetes/noobaa/noobaa-backing-store-crd.yaml

```diff
@@ -1,14 +1,46 @@
+###############################################################################
+#
+# CRD: NooBaaBackingStore
+#
+# WAIT !!! No need to create this manually, the operator does it.
```

Comment:
Thanks! I do have a question on the suggested security model - when the operator is responsible for a certain resource, say a CRD or a Role, it can update it on operator upgrade to match the operator version. How is this supposed to work in this secure model? What if the operator relies on a certain role to be applied in the new version? What if the CRD validation changes between versions?

Reply:
@guymguym The person that manages the operator is responsible for updating the RBAC, CRDs and other "parts" during upgrades, so the operator just runs a reconciliation loop every time it starts (e.g., as Ceph does when operators starts, a new disk is added).

If there are breaking changes to a CRD, the operator will take of migrating them to, e.g., a new CRD version (e.g., `v1alpha1` -> `v1alpha2`).
Searching for "migration", e.g., https://github.com/rook/rook/blob/818ad9c7ca4d352a9bed202e619d47b05f1a4dab/pkg/operator/ceph/cluster/migration.go, should give you an insight to how Ceph operator does it right now.